packages/consumption/src/modules/requests/itemProcessors/utility/validateAttributeMatchesWithQuery.ts

```diff
@@ -0,0 +1,198 @@
+        if (query.attributeCreationHints.tags !== undefined && query.attributeCreationHints.tags.length !== 0) {
+            if (attribute.tags === undefined || attribute.tags.length === 0 || !query.attributeCreationHints.tags.some((aQueriedTag) => attribute.tags!.includes(aQueriedTag))) {
+                return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The tags of the provided IdentityAttribute do not contain at least one queried tag."));
+            }
+        }
+    }
+
+    return ValidationResult.success();
+}
+
+function validateAttributeMatchesWithRelationshipAttributeQuery(
+    query: RelationshipAttributeQuery,
+    attribute: IdentityAttribute | RelationshipAttribute,
+    recipient: CoreAddress
+): ValidationResult {
+    if (!(attribute instanceof RelationshipAttribute)) {
+        return ValidationResult.error(
+            CoreErrors.requests.attributeQueryMismatch("The provided Attribute is not a RelationshipAttribute, but a RelationshipAttribute was queried.")
+        );
+    }
+
+    const recipientIsAttributeOwner = recipient.equals(attribute.owner);
+    const queriedOwnerIsEmpty = query.owner.equals("");
+
+    if (!queriedOwnerIsEmpty && !query.owner.equals(attribute.owner)) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute does not belong to the queried owner."));
+    }
+
+    if (queriedOwnerIsEmpty && !recipientIsAttributeOwner) {
+        return ValidationResult.error(
+            CoreErrors.requests.attributeQueryMismatch("You are not the owner of the provided RelationshipAttribute, but an empty string was specified for the owner of the query.")
+        );
+    }
+
+    if (query.key !== attribute.key) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute has not the queried key."));
+    }
+
+    if (query.attributeCreationHints.confidentiality !== attribute.confidentiality) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute has not the queried confidentiality."));
+    }
+
+    if (query.attributeCreationHints.valueType !== attribute.value.constructor.name) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute is not of the queried RelationshipAttribute Value Type."));
+    }
+
+    if (!(attribute.value instanceof Consent)) {
+        if (query.attributeCreationHints.title !== attribute.value.title) {
+            return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute has not the queried title."));
+        }
+
+        if (query.attributeCreationHints.description !== attribute.value.description) {
+            return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided RelationshipAttribute has not the queried description."));
+        }
```

Comment:
I don't think it's that easy to say. If, as the Sender of the Request, I would like to force a certain `description` for an Attribute or that the Attribute does not have any `description`, there should be a way to do this. I also understand the idea of giving the Recipient of the Request the option of changing certain values such as a `description`, but this can also be transferred to the value type of the Attribute, for example. IMHO first and foremost, the Sender should be allowed to decide how the Attribute is technically stored, and the Recipient then influences the actual `value` of the Attribute.